rtos/source/ThisThread.cpp

```diff
@@ -221,8 +221,7 @@ void ThisThread::sleep_for(uint32_t millisec)
 void ThisThread::sleep_for(Clock::duration_u32 rel_time)
 {
 #if MBED_CONF_RTOS_PRESENT
-    osStatus_t status = osDelay(rel_time.count());
-    MBED_ASSERT(status == osOK);
+    MBED_ASSERT(osOK == osDelay(rel_time.count()));
```

Comment:
I assume if `NDEBUG` is defined, this will become void and `osDelay` wont be executed?

Reply:
I think so. The question is whether the `MBED_ASSERT` macro is a strict assert (like the good-old `assert`), or a debug utility (which seems to be the case, as the statement isn't run if `NDEBUG`). If `osDelay()` is strictly required to work, we might want to use `assert()` instead, which gets retargeted to `mbed_assert_internal()` which always exists.

Reply:
```suggestion
    assert(osOK == osDelay(rel_time.count()));
```

Reply:
> we might want to use `assert()` instead,

No, that's eliminated in a release build in the same way.

> is it okay to use `mbed_assert_internal()` despite the name?

No, as it's internal, and it would leave the check in the release build.

You're just going to have to stick `(void) status;`, I think.

Reply:
Thanks for the reviews, I completly missed that!